cmd/clusterctl/client/repository/repository_memory_test.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func Test_memoryRepository(t *testing.T) {
+	metadata := `
+apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
+kind: Metadata
+releaseSeries:
+- major: 1
+  minor: 0
+  contract: v1alpha1
+- major: 2
+  minor: 0
+  contract: v1alpha3`
+
+	type want struct {
+		versions       []string
+		defaultVersion []byte
+		latestVersion  []byte
+	}
+	tests := []struct {
+		name        string
+		addTestData func(mr *MemoryRepository)
+		fileVersion string
+		want        want
+		wantErr     bool
+	}{
+		{
+			name: "Get the only release available from release directory",
+			addTestData: func(mr *MemoryRepository) {
+				mr.WithFile("v1.0.0", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v1.0.0", "components.yaml", []byte("v1.0.0"))
+				mr.WithPaths("", "components.yaml")
+			},
```

Comment:
```suggestion
			repo: NewMemoryRepository().
				WithFile("v1.0.0", "metadata.yaml", []byte(metadata)).
				WithFile("v1.0.0", "components.yaml", []byte("v1.0.0")).
				WithPaths("", "components.yaml"),
```
nit: Wouldn't it be simpler to just use a MemoryRepository as field?

Reply:
will do